src/UI/UIView.cpp

```diff
@@ -45,625 +51,897 @@ void UIView::draw_ui_callback()
+          m_animation_duration_to_string.clear();
+          m_camera_animation_duration_open = false;
+          ImGui::CloseCurrentPopup();
+        }
+        if (ImGui::Button("Cancel"))
+        {
+          m_camera_animation_duration_open = false;
+          m_animation_duration_to_string.clear();
+          ImGui::CloseCurrentPopup();
+        }
+        ImGui::EndPopup();
+      }
     }
-#ifdef WITH_TEEM
-    else if (helper::ends_with_string(fileDialog.selected_fn, ".nrrd")) {
-      std::vector<std::string> vals;
-      /*	vals.push_back(fileDialog.GetSelected().string());
-        promises.push_back(new std::promise<Volume*>);
-        futures.push_back(promises.back()->get_future());
-        threads.push_back(new std::thread(&VolumeVisualizationApp::loadVolume, this, vals, promises.back()));*/
+
+    // file loading
+    if (m_file_dialog_open)
+    {
+      ImGui::OpenPopup("Open File");
+      m_file_dialog_open = false;
     }
-#endif
-  }
 
+    if (fileDialog.showFileDialog("Open File", imgui_addons::ImGuiFileBrowser::DialogMode::OPEN, ImVec2(700, 310), m_file_extension_filter))
+    {
 
+      if (helper::ends_with_string(fileDialog.selected_fn, ".txt"))
+      {
+        VRDataLoader::load_txt_file(m_controller_app, fileDialog.selected_path);
+      }
+#ifdef WITH_TEEM
+      else if (helper::ends_with_string(fileDialog.selected_fn, ".nrrd"))
+      {
+        std::vector<std::string> vals;
+        /*	vals.push_back(fileDialog.GetSelected().string());
+          promises.push_back(new std::promise<Volume*>);
+          futures.push_back(promises.back()->get_future());
+          threads.push_back(new std::thread(&VolumeVisualizationApp::loadVolume, this, vals, promises.back()));*/
+      }
+#endif
+    }
 
+    if (m_file_dialog_save_dir)
+    {
+      ImGui::OpenPopup("Save File");
+      m_file_dialog_save_dir = false;
+      switch (m_current_save_modal)
+      {
+      case SAVE_SESSION:
+        m_save_session_dialog_open = false;
+        break;
+      case SAVE_TRFR_FNC:
+        m_save_trnfct_open = false;
+        break;
+      default:
+        break;
+      }
+    }
 
-  if (m_file_dialog_save_dir)
-  {
-    ImGui::OpenPopup("Save File");
-    m_file_dialog_save_dir = false;
-    switch (m_current_save_modal)
+    if (m_file_load_trnsf)
+    {
+      ImGui::OpenPopup("Load File");
+      m_file_load_trnsf = false;
+    }
+    if (saveDialogLoadTrnsFnc.showFileDialog("Save File", imgui_addons::ImGuiFileBrowser::DialogMode::SELECT,
+                                             ImVec2(700, 310)))
     {
-    case SAVE_SESSION:
-      m_save_session_dialog_open = false;
-      break;
-    case SAVE_TRFR_FNC:
-      m_save_trnfct_open = false;
-      break;
-    default:
-      break;
+      m_dir_to_save = saveDialogLoadTrnsFnc.selected_path;
+      switch (m_current_save_modal)
+      {
+      case SAVE_SESSION:
+        m_save_session_dialog_open = true;
+        break;
+      case SAVE_TRFR_FNC:
```

Comment:
I think this should be `SAVE_TRANSFER_FUNCTION` or `SAVE_TF`, etc - however you decided to abbreviate "transfer function" elsewhere

Reply:
renamed it  to SAVE_TRANSFER_FUNCTION